choreolib/src/main/native/include/choreo/lib/Choreo.h

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Choreo contributors
+
+#pragma once
+
+#include <frc/controller/PIDController.h>
+#include <frc/kinematics/ChassisSpeeds.h>
+#include <frc2/command/CommandPtr.h>
+#include <frc2/command/Requirements.h>
+
+#include <functional>
+#include <string>
+
+#include "ChoreoTrajectory.h"
+
+namespace choreolib {
+
+/// A type alias to constrain the controller function
+using ChoreoControllerFunction =
+    std::function<frc::ChassisSpeeds(frc::Pose2d, ChoreoTrajectoryState)>;
+
+/// A class that handles loading choreo trajectories and creating command
+/// factories for following trajectories
+class Choreo {
+ public:
+  /**
+   * Load a trajectory from the deploy directory. Choreolib expects .traj files
+   *  to be placed in src/main/deploy/choreo/[trajName].traj .
+   *
+   * @param trajName the path name in Choreo, which matches the file name in the
+   *  deploy directory. Do not include ".traj" here.
+   * @return the loaded trajectory, will throw runtime error if the file doesn't
+   *  exist
+   */
+  static ChoreoTrajectory GetTrajectory(std::string trajName);
```

Comment:
```suggestion
  static ChoreoTrajectory GetTrajectory(std::string_view trajName);
```
Add `#include <string_view>` to the top of the file as well.